dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -236,6 +238,38 @@ public Page<Event> getEvents(EventQueryParams queryParams, PageParams pageParams
     return new Page<>(events, pageParams, () -> getEventCount(queryParams));
   }
 
+  public List<Event> getEvents(
+      EventQueryParams queryParams, Map<String, Set<String>> psdesWithSkipSyncTrue) {
+    if (queryParams.isSynchronizationQuery()
+        && psdesWithSkipSyncTrue != null
+        && !psdesWithSkipSyncTrue.isEmpty()) {
+      queryParams = queryParams.withSkipSyncFiltering(psdesWithSkipSyncTrue);
+    }
+    return fetchEvents(queryParams, null);
+  }
+
+  public long countEvents(EventQueryParams queryParams) {
+    return getEventCount(queryParams);
```

Comment:
Maybe we could just make the getEventCount public?

Reply:
Yes, we can make it `public`, but it holds internal query logic that should nott be exposed. If made `public`, it might end up being used in multiple places so making it hard to change/refactor internal query logic.
But i am open to change it to `public` if we agree on it?

Reply:
i think you are right. Let me fix that.